src/libs/map-engine.ts

```diff
@@ -163,32 +163,27 @@ export default class MapEngine {
 
     public updateFlight(flight: Flight) {
         const {
-            geography: { latitude, longitude, direction, altitude },
+            geography: { latitude, longitude, direction },
             icaoNumber,
         } = flight;
 
         const enabled = true;
         const oldFlightInfo = this.flights[icaoNumber];
         const zoom = this.map.getZoom();
-
-        if (oldFlightInfo) {
-          if (altitude === 0) {
-              this.removeFlight(icaoNumber);
-          } else {
-              const marker = oldFlightInfo.marker;
-              google.maps.event.clearListeners(marker, 'click');
-              marker.addListener('click', () => this.openPopupForFlight(flight, marker));
-              if (this.icaoNumberToPopup.icaoNumber === icaoNumber && this.icaoNumberToPopup.popup) {
-                this.icaoNumberToPopup.popup.setContent(createPopup(flight));
-              }
-              createOrUpdateMarker(icaoNumber, longitude, latitude, { direction, enabled, marker, zoom });
-          }
-        } else if (altitude !== 0) {
-            // Handle new flight
-            const { marker } = createOrUpdateMarker(icaoNumber, longitude, latitude, { direction, enabled, zoom });
-            google.maps.event.addListener(marker!, 'click', () => this.openPopupForFlight(flight, marker!));
-            marker!.setMap(this.map);
-            this.flights[icaoNumber] = { flight, marker: marker! };
+        const marker = createOrUpdateMarker(
+          icaoNumber,
+          longitude,
+          latitude,
+          { direction, enabled, marker: oldFlightInfo?.marker, zoom },
+        ).marker!;
+        google.maps.event.clearListeners(marker, 'click');
+        google.maps.event.addListener(marker, 'click', () => this.openPopupForFlight(flight, marker));
+        if (this.icaoNumberToPopup.icaoNumber === icaoNumber && this.icaoNumberToPopup.popup) {
+          this.icaoNumberToPopup.popup.setContent(createPopup(flight));
+        }
+        if (!oldFlightInfo) {
```

Comment:
do you see any drawback in removing this check? Probably the next two lines are idempotent.

Reply:
No drawback, I made a test without this check and it was good. The only reason I kept the check is related to the performance. I made a little performance test with and without the `if` and it seems that without this check the system load is a bit higher but... honestly I'm not sure we can trust the result :|

Reply:
Nice. Let's prefer performance if it's measurable.

Reply:
after some other tests I decided to remove the check